app/src/androidTest/java/com/duckduckgo/app/browser/SubFrameUrlHandlerTest.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2019 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser
+
+import android.content.Intent
+import android.webkit.WebView
+import androidx.test.annotation.UiThreadTest
+import androidx.test.platform.app.InstrumentationRegistry.getInstrumentation
+import com.duckduckgo.app.browser.SpecialUrlDetector.UrlType.*
+import com.nhaarman.mockitokotlin2.mock
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+
+class SubFrameUrlHandlerTest {
```

Comment:
I can't see a `MainFrameUrlHandlerTest` class so I'm assuming the logic there has been tested somewhere else? If not, it might be a good idea to add them.

Reply:
Good point; these should be added. Created https://app.asana.com/0/414730916066338/1149174387975737 for adding them after this PR.